src/java/org/apache/cassandra/audit/AuditLogManager.java

```diff
@@ -197,8 +198,38 @@ public void logBatch(String batchTypeName, List<Object> queryOrIdList, List<List
         }
     }
 
-    public void reloadFilters()
+    public synchronized void disableAuditLog()
+    {
+        logger.info("Audit logging is disabled, stopping any existing loggers"); //debug
+        if (this.auditLogger != null)
+        {
+            this.auditLogger.stop();
+            this.auditLogger = null;
```

Comment:
I think this can race with ``log`` (which is not synchronized afaict) and cause a NPE. I think it's reasonable though to just catch NPE in ``log`` since that can only happen if the user has requested logging to stop.

Reply:
As per our offline discussion, instead of NPE handling in `log`, implemented `NoOpAuditLogger`.